src/c.rs

```diff
@@ -80,17 +69,57 @@ macro_rules! define_metrics_tests {
     }
 }
 
+#[cfg(test)]
+extern {
+    // We can't use `size_t` because we need to test that our
+    // definition of `size_t` is correct using this code! We use `u16`
+    // because even 8-bit and 16-bit microcontrollers have no trouble
+    // with it, and because `u16` is always as smaller or smaller than
+    // `usize`.
+    static GFp_int_align: u16;
+    static GFp_int_size: u16;
+    static GFp_long_align: u16;
+    static GFp_long_size: u16;
+    static GFp_size_t_align: u16;
+    static GFp_size_t_size: u16;
+    static GFp_int64_t_size: u16;
+    static GFp_int64_t_align: u16;
+    static GFp_uint64_t_size: u16;
+    static GFp_uint64_t_align: u16;
+    static GFp_int32_t_size: u16;
+    static GFp_int32_t_align: u16;
+    static GFp_uint32_t_size: u16;
+    static GFp_uint32_t_align: u16;
+    static GFp_int16_t_size: u16;
+    static GFp_int16_t_align: u16;
+    static GFp_uint16_t_size: u16;
+    static GFp_uint16_t_align: u16;
+    static GFp_int8_t_size: u16;
+    static GFp_int8_t_align: u16;
+    static GFp_uint8_t_size: u16;
+    static GFp_uint8_t_align: u16;
```

Comment:
So instead, I define them all beforehand. Maybe there are cleverer ways to conditionally define stuff in a macro?